experimental/dds/tree2/src/test/snapshots/utils.ts

```diff
@@ -0,0 +1,23 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { promises as fs, existsSync } from "fs";
+import { Serializable } from "@fluidframework/datastore-definitions";
+
+const numberOfSpaces = 4;
+
+export async function createSnapshot(path: string, data: Serializable): Promise<void> {
+	const dataStr = JSON.stringify(data, undefined, numberOfSpaces);
+	await fs.writeFile(path, dataStr);
+}
+
+export async function isEqualPastSnapshot(path: string, data: Serializable): Promise<boolean> {
```

Comment:
You'll get more informative test failures by giving this function assert semantics either: `verifySnapshotMatches(path: string, data: Serializable): Promise<void>` and then `assert.equal(dataStr, pastDataStr)` on the last line.